core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -318,7 +318,9 @@ private static String getCurrentExecVersion() {
     public static synchronized void persistInstallStatus(List<UpdateCenterJob> installingPlugins) {
         File installingPluginsFile = getInstallingPluginsFile();
 	if(installingPlugins == null || installingPlugins.isEmpty()) {
-		installingPluginsFile.delete();
+		if (!installingPluginsFile.delete()) {
+		    LOGGER.log(SEVERE, String.format("Failed to delete %s", installingPluginsFile.getAbsolutePath()));
```

Comment:
`Files` API as well. In such case the error cause will be propagated as exception